controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -503,6 +509,19 @@ func checkStaticPodReadyCondition(pod *corev1.Pod) error {
 	return nil
 }
 
+func checkNodeNoExecuteCondition(node *corev1.Node) error {
+	found := false
+	for _, taint := range node.Spec.Taints {
+		if taint.Key == corev1.TaintNodeUnreachable && taint.Effect == corev1.TaintEffectNoExecute {
+			found = true
```

Comment:
Based on what I read in #2469, it seems like we should be halting the upgrade if we **do** find this taint, right @detiber?

Reply:
That is correct, and this should accomplish that, with the exception that is already being tracked in https://github.com/kubernetes-sigs/cluster-api/issues/2430

Reply:
My bad, I misread the logic. Thanks!